docs/agents/spec-process.md

```diff
@@ -0,0 +1,58 @@
+Every change that affects more than one agent should be initiated via a change to the specification rather than creating a cross-agent issue.
+
+## Advantages
+
+- Ensures we have an up-to-date [spec](https://www.joelonsoftware.com/2000/08/09/the-joel-test-12-steps-to-better-code/) for agents.
+- Easier to keep track of what the current state of the proposal is.
+- Comments to a specific sentence of the spec can be made inline,
+  just like in code reviews.
+- If there are changes or amendments to a spec later on,
+  a PR to an existing spec is much cleaner than having to track down all preceding issues to find out what the current state is.
+- Easier to get a new agent (in-house or community) started.
+
+## Process
+
+1. Agents discussion issue \
+  If there's a lot of uncertainty about what the poposal is,
+  a discussion issue should be created.
+  In doubt, lean towards not creating a discussion issue and start with a draft PR.
+  Discussion issues should not have votes/agent checkboxes.
+  It's not expected that all agents participate in the initial discussions, although everyone is invited to do so.
+1. Draft PR \
+  It's fine if the first draft of the spec change even if details are unclear and need discussion.
+  The PR should include which agents are affected by this change.
+  Specifically, it should always be clear whether the RUM agent is affected by this.
+  This doesn't have to be determined yet when creating the draft PR,
+  but it is required before marking the PR as `Ready for Review`.
+1. Approval by at least one other agent \
+  This makes sure the change makes sense for not just one agent.
+  This doesn't ping all agent teams yet,
+  which only happens in the next step.
+  Of course,
+  every agent is more than welcome to contribute at any point in this process,
+  but they are not expected to actively keep track if it.
+1. Ready for Review \
+  This automatically requests review from all agents due to the [`CODEOWNERS`](https://github.com/elastic/apm/tree/master/.github/CODEOWNERS) file.
+  After a quorum of agents has approved, others have 1 week to veto by requesting changes.
```

Comment:
To this part:
> After a quorum of agents has approved

This sound very nice in theory, but I'm not totally sure it'll wok well in practice. I personally would not even be able to define what quorum of agents is off the top of my head. Why don't we just say "After at least 1 other agent approved" (or any other reasonable number). The draft needs to be approved by 1 other agent, this would be another one and probably the person who opens the PR also does it for a specific agent, so just by 1 additional approve here we'd already cover 3 agents at this point.

I know it's extremely simple, but I feel that's easier to follow and won't make a huge difference for the outcome.

Reply:
Good idea. Before changing I'd like to hear a few more opinions. @roncohen also raised some concerns about this part.

Reply:
Should we maybe require one more agent and a PM to approve? Then we'd have at least a total of 3 agent's and 1 PM's approval.

This makes sure PMs are always in the loop. Often times PMs would request a feature. If they don't see the exact proposal there's a chance the proposal doesn't do what they had in mind.
In other cases, a dev might have proposed a spec proposal based on a request form the community. In this case, it's also important to let PMs know about those changes.
Obviously, additional devs and leads can always approve or object at any given time.

The balance I'm trying to strike is to make sure everyone is in the loop while minimizing noise and not making the process too heavyweight.

Reply:
Seems to be the only outstanding question. I've changed it to what I've described in the previous comment. Happy to revisit though.

A counter-argument would be that this may introduce a bottleneck on the PM side. However, we already need their input for prioritization anyway.

Reply:
I think this is fine. This will force us a bit to have more discussions with PMs which is a good thing; especially in this case, because these are real world changes in agents, not only high level discussions. If this turns out to be a bottleneck we can still change it later.